test/extended/openstack/cpms.go

```diff
@@ -0,0 +1,129 @@
+package openstack
+
+import (
+	"context"
+	"strconv"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/stretchr/objx"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] ControlPlane MachineSet", func() {
+
+	defer g.GinkgoRecover()
+
+	var ctx context.Context
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+	var controlPlaneMachineSets []objx.Map
+
+	g.BeforeEach(func() {
+		ctx = context.Background()
+
+		g.By("preparing openshift dynamic client")
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		skipUnlessMachineAPIOperator(ctx, dc, clientSet.CoreV1().Namespaces())
+		controlPlaneMachineSets, err = getControlPlaneMachineSets(ctx, dc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("has role master", func() {
+		labels := controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.metadata.labels").Data().(map[string]interface{})
+		o.Expect(labels[machineLabelRole]).To(o.Equal("master"), "unexpected or absent role label")
```

Comment:
nit: maybe we can use this `ObjxMap` afterall, just to avoid adding even more failure modes to the bunch

```suggestion
		labels := controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.metadata.labels").ObjxMap()
		o.Expect(labels.Get(machineLabelRole).String()).To(o.Equal("master"), "unexpected or absent role label")
```

Reply:
Done